src/shared/parameter_v2/scripts/config_yaml_loader.py

```diff
@@ -0,0 +1,288 @@
+import os
+import sys
+
+import networkx
+import yaml
+from colorama import Fore, init
+import jsonschema
+
+from dynamic_parameter_schema import (
+    INCLUDE_DEF_SCHEMA,
+    PARAM_DEF_SCHEMA,
+    SUPPORTED_TYPES,
+)
+
+#######################################################################
+#                         Config Yaml Loader                          #
+#######################################################################
+
+
+class ConfigYamlLoader(object):
+    @staticmethod
+    def get_config_metadata(yaml_paths: list):
+        """Loads the yamls in the root config directory and verifies that the
+        requested configuration is valid. Then returns config_metadata dict
+        with a specified format capturing all the requested params and configs.
+
+        Example config directory:
+        .
+        └── config
+            ├── ai
+            │   ├── ai.yaml
+            │   ├── passing
+            │   │   └── passing.yaml
+            │   └── plays
+            │       ├── plays.yaml
+            │       ├── defense.yaml
+            │       └── offense.yaml
+            ├── sensor_fusion
+            │   └── sensor_fusion.yaml
+            └── tbots_config.yaml
+
+        A valid config directory maintains the following properties:
+
+        - All yamls are properly formed (i.e proper syntax, YAML Version 1.2)
+        - All parameter definitions abide by the dynamic parameter schema
+        - All include statements do NOT cause cycles in configs
+
+        :param yaml_paths: the path to all the config yamls
+        :type yaml_paths: list of str
```

Comment:
Just curious why we're taking a list if we just need the root directory? Just future-proofing?

Reply:
yeah essentially, the loader is not responsible for understanding the hierarchy of the files

so if we do change things, we can selectively give the loader a list of files. The input currently is essentially just the output of glob(**/*.yaml)